tests/components/abode/test_alarm_control_panel.py

```diff
@@ -0,0 +1,84 @@
+"""Tests for the Abode alarm control panel device."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import ATTR_DEVICE_ID
+from homeassistant.components.alarm_control_panel import DOMAIN as ALARM_DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    SERVICE_ALARM_ARM_AWAY,
+    SERVICE_ALARM_ARM_HOME,
+    SERVICE_ALARM_DISARM,
+    STATE_ALARM_DISARMED,
+)
+
+from .common import setup_platform
+
+
+async def test_entity_registry(hass):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, ALARM_DOMAIN)
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entry = entity_registry.async_get("alarm_control_panel.abode_alarm")
+    # Abode alarm device unique_id is the MAC address
+    assert entry.unique_id == "001122334455"
+
+
+async def test_automation_attributes(hass):
+    """Test the alarm control panel attributes are correct."""
+    await setup_platform(hass, ALARM_DOMAIN)
+
+    state = hass.states.get("alarm_control_panel.abode_alarm")
+    assert state.state == STATE_ALARM_DISARMED
+    assert state.attributes.get(ATTR_DEVICE_ID) == "area_1"
+    assert not state.attributes.get("battery_backup")
+    assert not state.attributes.get("cellular_backup")
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "Abode Alarm"
+    assert state.attributes.get(ATTR_SUPPORTED_FEATURES) == 3
+
+
+async def test_set_alarm_away(hass):
+    """Test the alarm control panel can be set to away."""
+    await setup_platform(hass, ALARM_DOMAIN)
+
+    with patch("abodepy.ALARM.AbodeAlarm.set_away") as mock_set_away:
+        await hass.services.async_call(
+            "alarm_control_panel",
+            SERVICE_ALARM_ARM_AWAY,
+            {ATTR_ENTITY_ID: "alarm_control_panel.abode_alarm"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
```

Comment:
Another piece I'm not entirely sure about. In the `hass.services.async_call`, `blocking=True` is passed which I would think is basically doing the same thing as `await hass.async_block_till_done()`. But when I tried removing it, CI tests seem to have issues.

Reply:
It depends on our case, if the service spawns another task in the event loop or if it does the call directly. The service will only block until the service handler is done. If there's another task involved that is scheduled after the service handler is done, we need to add the extra `async_block_till_done`.